conan/cache/cache.py

```diff
@@ -136,11 +136,20 @@ def _move_prev(self, old_pref: ConanReference, new_pref: ConanReference):
         except ReferencesDbTable.ReferenceAlreadyExist:
             # This happens when we create a recipe revision but we already had that one in the cache
             # we remove the new created one and update the date of the existing one
+            # TODO: cache2.0 locks
             self._db.delete_ref_by_path(old_path)
+            if os.path.exists(self._full_path(new_path)):
+                try:
+                    rmdir(self._full_path(new_path))
```

Comment:
I think this pattern we might want to make a common more robust, retry, capture exception ``conan.tools.files.rmdir``, but that can go in another PR.